package.json

```diff
@@ -18,12 +18,13 @@
   "types": "dist/index.d.ts",
   "main": "dist/index.js",
   "scripts": {
+    "clean": "for dir in dist node_modules typechain; do mv \"${dir}\" \"_${dir}\" 2>/dev/null; rm -rf \"_${dir}\" & done",
     "compile-zksync": "COMPILE_ZK=true yarn hardhat compile",
     "lint-contracts": "yarn solhint ./contracts/**/*.sol",
     "lint": "yarn prettier --list-different",
     "lint-fix": "yarn prettier --write",
     "prettier": "prettier .",
-    "build": "hardhat compile && tsc && rsync -a --include '*/' --include '*.d.ts' --exclude '*' ./typechain ./dist/",
+    "build": "[ -d typechain ] || yarn generate-contract-types && hardhat compile && tsc && rsync -a --include '*/' --include '*.d.ts' --exclude '*' ./typechain ./dist/",
```

Comment:
Should this be run regardless if the directory exists? What if the code has changed?